src/xtgeo/grid3d/_ecl_grid.py

```diff
@@ -428,19 +488,62 @@ def transform_xtgeo_coord_by_mapaxes(self, coord: np.ndarray):
 
         return coord
 
-    def xtgeo_coord(self):
+    def inverse_transform_xtgeo_coord_by_mapaxes(self, coord: np.ndarray):
+        """Inversely transforms xtgeo coord values by mapaxes.
+
+        The inverse operation of transform_xtgeo_coord_by_mapaxes.
+        """
+        x_point = self.mapaxes.x_line
+        y_point = self.mapaxes.y_line
+        origin = self.mapaxes.origin
+
+        x_axis = np.array(x_point) - origin
+        y_axis = np.array(y_point) - origin
+
+        x_unit = x_axis / np.linalg.norm(x_axis)
+        y_unit = y_axis / np.linalg.norm(y_axis)
+
+        coord[:, :, (0, 1)] -= np.array(origin)
+        coord[:, :, (3, 4)] -= np.array(origin)
+
+        inv_transform = np.linalg.inv(np.transpose([x_unit, y_unit]))
+
+        # The following index manipulation is
+        # an optimized version of
+
+        # nx, ny, _ = coord.shape
+        # for i in range(nx):
+        #    for j in range(ny):
+        #        coord[i, j, (0, 1)] = inv_transform @ coord[i, j, (0, 1)]
+        #        coord[i, j, (3, 4)] = inv_transform @ coord[i, j, (3, 4)]
+        coord[:, :, (0, 1)] = (
+            inv_transform[np.newaxis, np.newaxis, :, :]
+            @ coord[:, :, (0, 1), np.newaxis]
+        )[:, :, :, 0]
+        coord[:, :, (3, 4)] = (
+            inv_transform[np.newaxis, np.newaxis, :, :]
+            @ coord[:, :, (3, 4), np.newaxis]
+        )[:, :, :, 0]
+
+        return coord
+
+    def xtgeo_coord(self, coordinates=GridRelative.MAP):
```

Comment:
I would expect `coordinates` to accept numbers, not an enum instance. Rename to `coordinates_type` or something more appropriate?

Reply:
I renamed it to `relative_to` as the coordinate_type name is already taken for cartesian vs radial.